### PR TITLE
BarChart: show x tick picker

### PR DESCRIPTION
**Pull request by ryantxu**
In #43257 we added a feature to skip some x axis labels, but the UI to control it is a little funky.  In this PR we are adding a custom T-shirt size control:

Peek 2022-01-05 16-51

Previously as a slider:
image

### CODE REVIEW

**Comment by ryantxu** [comment]
@dprokop -- this is currently failing with cue/schema issues.  I'll pull the devenv dashboard into its own PR if the cue issues are not fixed before this is ready to commit

File: public/app/plugins/panel/barchart/TickSpacingEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    description: 'Require 300px spacing',
+  },
+];
+
+export const TickSpacingEditor: React.FC<StandardEditorProps<number, any>> = (props) => {
+  let value = props.value ?? 0;
+  const isRTL = value < 0;
+  if (isRTL) {
+    value *= -1;
+  }
```
Comment (dprokop) [inline public/app/plugins/panel/barchart/TickSpacingEditor.tsx:33]: Nothing that blocks this PR from being merged, but I'm not a huge fan of relying on the value when configuring RTL.

**Review by dprokop — APPROVED** [review]
looks good!

File: public/app/plugins/panel/barchart/TickSpacingEditor.tsx (hunk 1 above)
Comment (ryantxu) [reply public/app/plugins/panel/barchart/TickSpacingEditor.tsx:33]: > but I'm not a huge fan of relying on the value when configuring RTL

Any alternative ideas?  everything I can think of adds up to the same thing

**Pull request merged**